File: State/dart/Canvas.dart
import 'UIControl.dart';

class Canvas {
  UIControl _currentTool;

  UIControl get currentTool => _currentTool;

  set currentTool(UIControl currentTool) {
    _currentTool = currentTool;
  }

  Canvas({
    required currentTool,
  }) : _currentTool = currentTool;

  void mouseDown() {
    currentTool.mouseDown();
  }

  void mouseUp() {
    currentTool.mouseUp();
  }
}

File: State/python/Canvas.py
from UIControl import UIControl


class Canvas:
    def __init__(self, currentTool: UIControl) -> None:
        self.__currentTool = currentTool

    def setCurrentTool(self, currentTool: UIControl) -> None:
        self.__currentTool = currentTool

    def mouseDown(self) -> None:
        self.__currentTool.mouseDown()

    def mouseUp(self) -> None:
        self.__currentTool.mouseUp()
